pkg/didmethod/peer/did.go

```diff
@@ -30,7 +32,59 @@ func newDid(doc *did.Doc) (string, error) {
 	if doc.PublicKey == nil || doc.Authentication == nil {
 		return "", errors.New("the genesis version must include public keys and authentication")
 	}
+	numBasis, err := hashString(doc)
+	if err != nil {
+		return "", err
+	}
+	messageIdentifier := []string{peerPrefix, numAlgo, encAlgo, "-", numBasis}
+	peerDID := strings.Join(messageIdentifier, "")
+
+	return peerDID, nil
+}
+
+func validate(doc *did.Doc) error {
```

Comment:
This needs some godocs explaining what it does and, most importantly, the requirements it has on the input (it works only on the genesis version of the peer DID doc)

Reply:
added comments

Reply:
Sorry - I think we should rename this function to `validateDid` (below I suggested to remove the existing "validateDid" function) and I think a simpler comment could be:

```
validateDid checks the format of the doc's DID and checks that the DID's 'namestring' matches against its numeric basis as per the Namestring Generation Method: https://openssi.github.io/peer-did-method-spec/index.html#namestring-generation-method
Note: this check should be done only on the resolved variant of the genesis version of Peer DID documents.
```